tensorflow_tts/processor/base_processorv2.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import abc
+import os
+import json
+
+from typing import Union
+
+
+@dataclass
+class ExamplePrepro(abc.ABC):
+    root_dir: str
+    symbols: list
+    speakers_map = {}
+    train_f_name: str = "train.txt"
+    delimiter: str = "|"
+    positions = {"file": 0, "text": 1, "speaker_name": 2}  # positions of file,text,speaker_name after split line
+    f_extension: str = ".wav"
+    extra_tokens = {"unk": "[UNK]", "pad": "[PAD]", "eos": "[EOS]", "bos": "[BOS]"}
+    save_mapper: bool = False
+
+    # extras
+    items = []
+    symbol_to_id = {}
+    id_to_symbol = {}
+
+    def __post_init__(self):
+        self.create_items()
+        self.create_speaker_map()
+        self.reverse_speaker = {v: k for k, v in self.speakers_map.items()}
+        self.create_symbols()
+        if self.save_mapper:
+            self.__save_mapper()
+
+    def __getattr__(self, name: str) -> str:
+        if "_id" in name:  # map extra token to id
+            return self.symbol_to_id[self.extra_tokens[name.split("_id")[0]]]
+        return self.extra_tokens[name]  # map extra token to value
+
+    def create_speaker_map(self):
+        """
+        Create speaker map for dataset
+        """
+        sp_id = 0
+        for i in self.items:
+            speaker_name = i[-1]
+            if speaker_name not in self.speakers_map:
+                self.speakers_map[speaker_name] = sp_id
+                sp_id += 1
+
+    def get_speaker_id(self, name: str) -> int:
+        return self.speakers_map[name]
+
+    def get_speaker_name(self, speaker_id: int) -> str:
+        return self.speakers_map[speaker_id]
+
+    def create_symbols(self):
+        self.symbols = self.symbols + list(self.extra_tokens.values())
+        self.symbol_to_id = {s: i for i, s in enumerate(self.symbols)}
+        self.id_to_symbol = {i: s for i, s in enumerate(self.symbols)}
+
+    def create_items(self):
```

Comment:
This function should be consider as abstractmethod ?. The logic of this function seems not general for all dataset format ?

Reply:
It is you need to preprocess your dataset to some .txt file with this format and it'll be a lot easier to do then 

file_path | text. | speaker

or 

speaker | text. | file_path   

etc.


So as method it can be abstract but for most cases it'll be used as basic method (most other repos also use this format in their examples (nvidia, flowtron, taco2 etc.) so it'll be easier to move from other repos to our using some sort of standard train.txt format 👍 And if people need to change it they'll just overload method ( thanks to post_init in dataclasses we don't need to considered overloading in __init__ :P)

Example overload ->


```python
@dataclass
class LJ(ExamplePrepro):

    def get_one_sample(self, item):
        sample = {
            "raw_text": None,
            "text_ids": None,
            "audio": None,
            "utt_id": None,
            "speaker_name": None,
            "rate": None,
        }
        return sample

    def create_items(self):
        for i in range(2):
            self.items.append(["MY OWN CREATE ITEM", "MY PATH", "ME!"])

    def text_to_sequence(self, text):
        return ["0"]

e = LJ("files", list(string.ascii_lowercase))
print(e.items)  => # [['MY OWN CREATE ITEM', 'MY PATH', 'ME!'], ['MY OWN CREATE ITEM', 'MY PATH', 'ME!']]
```